### PR TITLE
Add RPCs to start/stop sealing by the miner

### PR DESCRIPTION
**Pull request by jhs7jhs**
These RPCs are going to be added ~~for debugging purpose~~. #640 

- `devel_startSealing`
Starts and enables sealing blocks by the miner.
Also, the command will start sealing immediately even if there's no parcel in the memory pool.
It should be ensured that the option `force_sealing` is `true` in the miner config, to seal a block with no parcels.
- `devel_stopSealing`
Stops and disables sealing blocks by the miner.

### CODE REVIEW

File: core/src/consensus/solo/mod.rs [hunk 1]
```diff
@@ -59,12 +59,13 @@ where
...
     }
 
-    fn generate_seal(&self, block: &M::LiveBlock, _parent: &M::Header) -> Seal {
-        if block.parcels().is_empty() {
-            Seal::None
-        } else {
-            Seal::Regular(Vec::new())
-        }
+    fn generate_seal(&self, _block: &M::LiveBlock, _parent: &M::Header) -> Seal {
+        // if block.parcels().is_empty() {
```
Comment (kseo) [inline core/src/consensus/solo/mod.rs]: Delete commented code.

File: core/src/miner/miner.rs [hunk 2]
```diff
@@ -95,6 +96,8 @@ pub struct Miner {
     engine: Arc<CodeChainEngine>,
     options: MinerOptions,
 
+    devel_sealing_enabled: AtomicBool,
```
Comment (kseo) [inline core/src/miner/miner.rs]: Remove `devel_` prefix.

File: core/src/miner/miner.rs [hunk 3]
```diff
@@ -749,4 +753,22 @@ impl MinerService for Miner {
     fn future_parcels(&self) -> Vec<SignedParcel> {
         self.mem_pool.read().future_parcels()
     }
+
+    fn start_sealing<C: MiningBlockChainClient>(&self, client: &C) {
+        cdebug!(MINER, "Start Sealing");
```
Comment (kseo) [inline core/src/miner/miner.rs]: Sealing -> sealing

File: core/src/miner/miner.rs [hunk 4]
```diff
@@ -749,4 +753,22 @@ impl MinerService for Miner {
...
+        self.devel_sealing_enabled.store(true, Ordering::Relaxed);
+        // ------------------------------------------------------------------
+        // | NOTE Code below requires mem_pool and sealing_queue locks.     |
+        // | Make sure to release the locks before calling that method.     |
+        // ------------------------------------------------------------------
+        if self.parcel_reseal_allowed() {
+            cdebug!(MINER, "Update Sealing");
```
Comment (kseo) [inline core/src/miner/miner.rs]: Sealing -> sealing

File: core/src/miner/miner.rs [hunk 5]
```diff
@@ -749,4 +753,22 @@ impl MinerService for Miner {
...
+            self.update_sealing(client);
+        }
+    }
+
+    fn stop_sealing(&self) {
+        cdebug!(MINER, "Stop Sealing");
```
Comment (kseo) [inline core/src/miner/miner.rs]: Sealing -> sealing

File: core/src/miner/mod.rs [hunk 6]
```diff
@@ -112,6 +112,12 @@ pub trait MinerService: Send + Sync {
 
     /// Get a list of all future parcels.
     fn future_parcels(&self) -> Vec<SignedParcel>;
+
+    /// Start sealing for debugging.
```
Comment (kseo) [inline core/src/miner/mod.rs]: Remove "for debugging".

File: core/src/miner/mod.rs [hunk 7]
```diff
@@ -112,6 +112,12 @@ pub trait MinerService: Send + Sync {
...
+    fn start_sealing<C: MiningBlockChainClient>(&self, client: &C);
+
+    /// Stop sealing for debugging.
```
Comment (kseo) [inline core/src/miner/mod.rs]: Remove "for debugging".

File: spec/JSON-RPC.md [hunk 8]
```diff
@@ -1758,3 +1758,52 @@ Response Example
   "id":null
 }
 ```
+
+## devel_startSealing
+Starts and Enable sealing parcels by the miner.
```
Comment (kseo) [inline spec/JSON-RPC.md]: * Enable -> enables
* parcels -> blocks

File: spec/JSON-RPC.md [hunk 9]
```diff
@@ -1758,3 +1758,52 @@ Response Example
...
+```
+{
+  "jsonrpc":"2.0",
+  "result":null,
+  "id":null
+}
+```
+
+## devel_stopSealing
+Stops and Disable sealing parcels by the miner.
```
Comment (kseo) [inline spec/JSON-RPC.md]: Disable -> disables

File: test/src/basic.test.ts [hunk 10]
```diff
@@ -45,8 +45,11 @@ describe("solo - 1 instance", () => {
     }));
     const invoice = await instance.sdk.rpc.chain.getParcelInvoice(parcelHash);
     expect(invoice).toEqual({ success: true });
+    // instance.sdk.rpc.chain.getPendingParcels()
   });
 
+
```
Comment (kseo) [inline test/src/basic.test.ts]: Avoid unnecessary changes.

File: test/src/basic.test.ts [hunk 11]
```diff
@@ -45,8 +45,11 @@ describe("solo - 1 instance", () => {
     }));
     const invoice = await instance.sdk.rpc.chain.getParcelInvoice(parcelHash);
     expect(invoice).toEqual({ success: true });
+    // instance.sdk.rpc.chain.getPendingParcels()
```
Comment (kseo) [inline test/src/basic.test.ts]: Why?

File: codechain/config/presets/config.dev.toml [hunk 12]
```diff
@@ -9,7 +9,7 @@ disable = false
 mem_pool_mem_limit = 4 # MB
 mem_pool_size = 8192
 notify_work = []
-force_sealing = false
+force_sealing = true
```
Comment (kseo) [inline codechain/config/presets/config.dev.toml]: Why?

**Comment by jhs7jhs** [comment]
@kseo Fixed.

File: spec/JSON-RPC.md [hunk 13]
```diff
@@ -1758,3 +1760,51 @@ Response Example
   "id":null
 }
 ```
+
+## devel_startSealing
+Starts and enabls sealing parcels by the miner.
```
Comment (kseo) [inline spec/JSON-RPC.md]: * enabls -> enables
* parcels -> blocks

File: spec/JSON-RPC.md [hunk 14]
```diff
@@ -1758,3 +1760,51 @@ Response Example
...
+```
+{
+  "jsonrpc":"2.0",
+  "result":null,
+  "id":null
+}
+```
+
+## devel_stopSealing
+Stops and disables sealing parcels by the miner.
```
Comment (kseo) [inline spec/JSON-RPC.md]: parcels -> blocks

**Comment by jhs7jhs** [comment]
@kseo Sorry for typos. Done.
